From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nick Turner <nic@amazon.com>
Date: Tue, 9 Nov 2021 18:50:19 +0000
Subject: [PATCH] --EKS-PATCH-- [AWS] Set max results if its not set

Cherry-pick of upstream Kubernetes:
https://github.com/kubernetes/kubernetes/pull/106280, which is
included in Kubernetes 1.24.

* If max results is not set and instance IDs are not provided for the
  describe instances call in the aws cloud provider, set max results.
* This prevents an expensive call against the EC2 API, which can
  result in timeouts.

Signed-off-by: Kirsten Schumy <ksschumy@amazon.com>
---
 staging/src/k8s.io/legacy-cloud-providers/aws/aws.go | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go b/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go
index b470d585f7e..172952369d5 100644
--- a/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go
+++ b/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go
@@ -991,6 +991,13 @@ func (s *awsSdkEC2) DescribeInstances(request *ec2.DescribeInstancesInput) ([]*e
 	results := []*ec2.Instance{}
 	var nextToken *string
 	requestTime := time.Now()
+
+	if request.MaxResults == nil && request.InstanceIds == nil {
+		// MaxResults must be set in order for pagination to work
+		// MaxResults cannot be set with InstanceIds
+		request.MaxResults = aws.Int64(1000)
+	}
+
 	for {
 		response, err := s.ec2.DescribeInstances(request)
 		if err != nil {
